server/tm_cmds.go

```diff
@@ -49,31 +47,24 @@ func ShowValidatorCmd() *cobra.Command {
 			cfg := serverCtx.Config
 
 			privValidator := pvm.LoadOrGenFilePV(cfg.PrivValidatorKeyFile(), cfg.PrivValidatorStateFile())
-			valPubKey, err := privValidator.GetPubKey()
+			pk, err := privValidator.GetPubKey()
 			if err != nil {
 				return err
 			}
-
-			output, _ := cmd.Flags().GetString(cli.OutputFlag)
-			if strings.ToLower(output) == "json" {
-				return printlnJSON(valPubKey)
-			}
-
-			pubkey, err := cryptocodec.FromTmPubKeyInterface(valPubKey)
+			sdkPK, err := cryptocodec.FromTmPubKeyInterface(pk)
 			if err != nil {
 				return err
 			}
-			pubkeyBech32, err := sdk.Bech32ifyPubKey(sdk.Bech32PubKeyTypeConsPub, pubkey)
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			bz, err := clientCtx.JSONMarshaler.MarshalInterfaceJSON(sdkPK)
 			if err != nil {
 				return err
 			}
-
-			fmt.Println(pubkeyBech32)
+			fmt.Println(string(bz))
 			return nil
 		},
 	}
 
-	cmd.Flags().StringP(cli.OutputFlag, "o", "text", "Output format (text|json)")
```

Comment:
This is actually a breaking change. Either add an entry to the changelog, or we could not make it a breaking change by:
- `--json`: ouput proto JSON
- `--text`: output sdkPk.String()